### PR TITLE
322 skills selection to multiselect

### LINKED ISSUE #322: Turn skills selection into multi-select

### PR DESCRIPTION
**Pull request by katmartel**
I figured I should make a PR for this now so we can discuss this as a team during our next meeting. So far, I've only modified the Mentee form to be a multiselect so that it can be compared to the Mentor form's separate select fields. 

Pros to Multiselect:
- The user can't select the same skill more than once.
- Subjectively, I think it looks better.

Cons:
- I haven't been able to find a way to limit the number of tags that can be selected (the client had suggested 3 skills).

What are your guys' thoughts on the multiselect, and are there any suggestions for limiting the tags?

### CODE REVIEW

File: src/front-end/src/components/MenteeForm.js [hunk 1]
```diff
@@ -67,6 +73,11 @@ class MenteeForm extends Component {
   }
 
   render() {
+    const { selectedItems } = this.state;
+    const filteredOptions = learnableSkills.filter(
+      (o) => !selectedItems.includes(o),
+    );
+
```
Comment (jpabadir) [inline src/front-end/src/components/MenteeForm.js]: Why don't we just assume nothing is selected and just pass learnableSkills?

File: src/front-end/src/components/MenteeForm.js [hunk 2]
```diff
@@ -128,54 +139,29 @@ class MenteeForm extends Component {
...
-                rules={[
-                  {
-                    required: true,
-                    message: 'what is the last skill you want to learn?',
-                  },
-                ]}
+              <Select
+                mode="multiple"
+                value={selectedItems}
+                onInput={this.handleSkillsChange}
```
Comment (jpabadir) [inline src/front-end/src/components/MenteeForm.js]: I don't think onInput works for selects. The handleSkillsChange method isn't getting called. onChange is what we want.

File: src/front-end/src/components/MenteeForm.js (hunk 1 above)
Comment (jpabadir) [reply src/front-end/src/components/MenteeForm.js]: Update: I deleted this when doing the two things I describe in my PR review description. Let me know if I made a mistake in which case I'm happy to add it back.

**Review by jpabadir — APPROVED** [review]
Great PR! I made a couple of commits to fix the bug I described in my comments, as well as to restrict the number of options which can be selected, since you asked for suggestions with that. For reference, the code I used is: 

```
              <Select
                mode="multiple"
                value={selectedItems}
                onChange={this.handleSkillsChange}
              >
                {learnableSkills.map((item) => (
                  <Option
                    key={item}
                    value={getCamelCase(item)}
                    disabled={
                      selectedItems.length >= 3 &&
                      !selectedItems.includes(getCamelCase(item))
                    }
                  >
                    {item}
                  </Option>
                ))}
              </Select>
```

**Review by aleksssk — APPROVED** [review]
Nice job!

File: src/front-end/src/components/MenteeForm.js (hunk 1 above)
Comment (katmartel) [reply src/front-end/src/components/MenteeForm.js]: Thanks for catching that!! Originally, I was going to try a different variation of the multiselect and completely forgot to remove that chunk of code when I changed it, sorry!

**Comment by katmartel** [comment]
> Great PR! I made a couple of commits to fix the bug I described in my comments, as well as to restrict the number of options which can be selected, since you asked for suggestions with that. For reference, the code I used is:
> 
> ```
>               <Select
>                 mode="multiple"
>                 value={selectedItems}
>                 onChange={this.handleSkillsChange}
>               >
>                 {learnableSkills.map((item) => (
>                   <Option
>                     key={item}
>                     value={getCamelCase(item)}
>                     disabled={
>                       selectedItems.length >= 3 &&
>                       !selectedItems.includes(getCamelCase(item))
>                     }
>                   >
>                     {item}
>                   </Option>
>                 ))}
>               </Select>
> ```

Thanks so much for working on that, JP! :)

**Review by wrc9817 — APPROVED** [review]

**Pull request merged**